File: swift/Tests/WarehouseTests/QuantityTests.swift
import Testing
import Warehouse

@Suite("Quantity")
struct QuantityTests {
  @Test func acceptsStrictlyPositiveValue() throws {
    #expect(try Quantity(1).value == 1)
  }

  @Test func rejectsZero() {
    expectInvalidOrder({ _ = try Quantity(0) }, contains: "strictly positive")
  }

  @Test func rejectsNegative() {
    expectInvalidOrder({ _ = try Quantity(-3) }, contains: "strictly positive")
  }

  @Test func rejectsAboveMax() {
    expectInvalidOrder({ _ = try Quantity(quantityMax + 1) }, contains: "exceeds")
  }

  @Test func acceptsSharedMaximum() throws {
    #expect(try Quantity(quantityMax).value == quantityMax)
  }
}

File: csharp/tests/Warehouse.UnitTests/QuantityTests.cs
using Warehouse.Domain;

namespace Warehouse.UnitTests;

public sealed class QuantityTests
{
    [Fact]
    public void AcceptsStrictlyPositiveValue()
    {
        Assert.Equal(1, new Quantity(1).Value);
    }

    [Fact]
    public void AcceptsSharedMaximum()
    {
        Assert.Equal(Quantity.MaxValue, new Quantity(Quantity.MaxValue).Value);
    }

    [Fact]
    public void RejectsZero()
    {
        InvalidOrderException error = Assert.Throws<InvalidOrderException>(() => new Quantity(0));
        Assert.Contains("strictly positive", error.Message, StringComparison.Ordinal);
    }

    [Fact]
    public void RejectsNegative()
    {
        InvalidOrderException error = Assert.Throws<InvalidOrderException>(() => new Quantity(-3));
        Assert.Contains("strictly positive", error.Message, StringComparison.Ordinal);
    }
}
